core/src/main/java/com/linecorp/armeria/common/PropagatingContextAwareExecutorService.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2021 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.common;
+
+import java.util.concurrent.ExecutorService;
+
+import javax.annotation.Nullable;
+
+import com.google.common.base.MoreObjects;
+
+final class PropagatingContextAwareExecutorService
+        extends AbstractContextAwareExecutorService<ExecutorService> {
+    PropagatingContextAwareExecutorService(ExecutorService executor) {
+        super(executor);
+    }
+
+    @Override
+    @Nullable
+    public RequestContext contextOrNull() {
```

Comment:
I think it's good to have the `@Nullable` here as well for clarity. Happy to remove `public`.

Reply:
As you know, Armeria is non-null by default. We do not annotate a method with `@Nullable` if the super method is already nullable.
We only override `@Nullable` with `@NonNull` we want to change the trait in the child class or interface.

Reply:
Hmm ok. Precisely because armeria is non-null by default, I think it would be a better policy for nullable methods to always be annotated `@Nullable`, but that's very much out of scope for this PR.

Reply:
Yeah, it's inherited, but I think it's better leaving it for readability. I thought IntelliJ adds it automatically anyway.